frontend/server/src/Controllers/Problem.php

```diff
@@ -4716,36 +4719,44 @@ public static function getProblemEditDetailsForSmarty(
         $groupAdmins = \OmegaUp\DAO\GroupRoles::getProblemAdmins($problem);
 
         $details = self::getProblemEditDetails($problem, $r->identity);
+        $extraInfo = [
+            'statusSuccess' => false,
+            'admins' => $admins,
+            'groupAdmins' => $groupAdmins,
+            'tags' => $tags,
+            'problemLevel' => \OmegaUp\DAO\ProblemsTags::getProblemLevel(
+                $problem
+            ),
+            'publicTags' => \OmegaUp\Controllers\Tag::getPublicTags(),
+            'levelTags' => \OmegaUp\Controllers\Tag::getLevelTags(),
+            'selectedTags' => $selectedTags,
+            'selectedPublicTags' => \OmegaUp\DAO\ProblemsTags::getTagsForProblem(
+                $problem,
+                !\OmegaUp\Authorization::canEditProblem($r->identity, $problem),
+                true
+            ),
+            'selectedPrivateTags' => \OmegaUp\DAO\ProblemsTags::getTagsForProblem(
+                $problem,
+                !\OmegaUp\Authorization::canEditProblem($r->identity, $problem),
+                false
+            ),
```

Comment:
a ver, qué te parece lo siguiente para que a `getTagsForProblem()` no tenga que importarle si el usuario tiene permisos:

```suggestion
            'selectedPublicTags' => \OmegaUp\DAO\ProblemsTags::getTagsForProblem(
                $problem,
                true
            ),
            'selectedPrivateTags' => (
                \OmegaUp\Authorization::canEditProblem($r->identity, $problem) ?
                \OmegaUp\DAO\ProblemsTags::getTagsForProblem(
                    $problem,
                    false
                ) : []
            ),
```